src/paymentservice/package.json

```
@@ -22,6 +22,10 @@
     "@opentelemetry/exporter-trace-otlp-grpc": "0.33.0",
     "@opentelemetry/exporter-metrics-otlp-grpc": "0.33.0",
     "@opentelemetry/sdk-node": "0.33.0",
+    "@opentelemetry/resource-detector-alibaba-cloud": "^0.27.3",
+    "@opentelemetry/resource-detector-aws": "^1.2.1",
+    "@opentelemetry/resource-detector-container": "^0.2.1",
+    "@opentelemetry/resource-detector-gcp": "^0.27.4",
```

Let's be consistent with pinning to exact version numbers

```suggestion
    "@opentelemetry/resource-detector-alibaba-cloud": "0.27.3",
    "@opentelemetry/resource-detector-aws": "1.2.1",
    "@opentelemetry/resource-detector-container": "0.2.1",
    "@opentelemetry/resource-detector-gcp": "0.27.4",
```